pkg/azure/ipam/node.go

```diff
@@ -48,6 +48,9 @@ func (n *Node) UpdatedNode(obj *v2.CiliumNode) {
 // resource with Azure specific information
 func (n *Node) PopulateStatusFields(k8sObj *v2.CiliumNode) {
 	k8sObj.Status.Azure.Interfaces = []types.AzureInterface{}
+
+	n.manager.mutex.RLock()
+	defer n.manager.mutex.RUnlock()
```

Comment:
I think I attempted a fix like this before, and the deadlock detector ended up flagging this as "inconsistent locking" with another lock inside `ForeachInterface`.

Reply:
Thanks for the heads-up. I didn‘t see any such message in my local testing but will keep an eye on the CI tests.